_reference/05_the_view_macro.adoc

```diff
@@ -0,0 +1,252 @@
+= The view! macro
+:page-permalink: /view-macro/
+
+== Introduction
+
+The `view!` macro is used within the `#[widget]` implementation.
+It allows us to describe the widget hierarchy in a declarative manner.
+It's an alternative to a glade file.
+
+[source,rust]
+----
+    view! {
+        gtk::Window {
+            gtk::Box {
+                orientation: Vertical,
+                gtk::Button {
+                    clicked => Msg::Increment,
+                    label: "+",
+                },
+                gtk::Label {
+                    text: &self.model.counter.to_string(),
+                },
+                gtk::Button {
+                    clicked => Msg::Decrement,
+                    label: "-",
+                },
+            },
+            delete_event(_, _) => (Msg::Quit, Inhibit(false)),
+        }
+    }
+----
+
+Within the `view!`, we can declare:
+
+- <<Widgets and their hierarchy>>,
+- <<Set widget properties>>,
+- <<Bind events>>
+
+== Widgets and their hierarchy
+
+=== Declare widgets
+
+[source,rust]
+----
+    view! {
+        gtk::Window {
+            gtk::Box {
+                gtk::Button {
+                    label: "+",
+                },
+                gtk::Button {
+                    label: "-",
+                },
+            },
+        }
+    }
+----
+
+In this setup, we declared a gtk window, containing a gtk box, containing two gtk buttons.
+We also set the `label` property of each button.
+Everytime we nest a widget within the `{}` of another widget, we set up a hierarchical parent/child or container/child relationship.
+
+The widgets contained in the `view` can be gtk widgets as well as relm widgets.
+
+=== Relm widgets
+
+Here is an example of adding a relm widget in a view hierarchy:
+
+
+[source,rust]
+----
+    SearchView((None, SearchItemsType::All, OperationMode::ItemActions)) {
+    }
+----
+
+In this case, `SearchView` is the name of the relm component, and in the brackets we have all the component's model parameters.
+
+As a reminder, when we declare a relm component, we must define its `fn model()` function. That function takes as as second parameter the model parameters.
+For instance, in this case:
+
+[source,rust]
+----
+    fn model(
+        relm: &relm::Relm<Self>,
+        params: (
+            Option<String>,
+            SearchItemsType,
+            OperationMode,
+        ),
+    ) -> Model {
+----
+
+The component model parameter has three parameters, which we pass through a tuple when building the `view!` hierarchy.
+
+=== Widget attributes
+
+We can also attach attributes to widgets. For instance:
+
+[source,rust]
+----
+    view! {
+        #[name="toplevel"]
+        #[style_class="mainwin"]
+        gtk::Window {
+        }
+    }
+----
+
+Let's now review the possible widget attributes.
+
+- <<The 'name' widget attribute>>,
```

Comment:
You need to update those as well since you updated the titles.

Reply:
good point, should be fixed now!